core/liblwm2m.c

```diff
@@ -328,3 +374,32 @@ int lwm2m_step(lwm2m_context_t * contextP,
 
     return 0;
 }
+
+#ifdef LWM2M_CLIENT_MODE
+int lwm2m_start(lwm2m_context_t * contextP)
```

Comment:
Reviewing your change, I come to realize I don't like the lwm2m_start() API...
Before its only job was to retrieve servers. Something that should be done by lwm2m_configure() or at the end of a bootstrap.
Now it is used as a reset. The client should do this by closing the lwm2m_context_t and create a new one configured with the same objects. The close callbacks for objects should also be removed. Objects data are the responsability of the client.

Your thoughts ?